### PR TITLE
wip: optimi

### PR DESCRIPTION
**Pull request by bghira**
it doesn't work yet. this is for other developers to review and test.

- [x] import optimi configs as defaults
- [x] map old --parameters to their new args
- [x] removed adafactor
- [x] removed dadaptation
- [x] removed prodigy
- [x] recommend adafactor -> stableadamw
- [x] enable gradient release by default on all optims
- [x] handle training loop changes eg. remove adafactor-specific avoidance of gradient clipping in favour of optimi-aware decision
- [x] update model card logic to reference correct optimiser and config
- [x] update documentation to indicate changes to optimiser configurations
- [x] run memory benchmarks to measure savings

### CODE REVIEW

**Comment by bghira** [comment]
cc @sayakpaul

**Comment by bghira** [comment]
nope, it's really not going to work. quanto just doesn't work with Optimi

**Comment by bghira** [comment]
not only do they not work with quanto, they also seem to require autocast despite claiming not to

**Comment by bghira** [comment]
i figured out how to make it work with quanto and now rank 128 training consumes same memory as rank 4 training, so that's cool
